addons/product/models/res_partner.py

```diff
@@ -30,9 +30,17 @@ def _inverse_product_pricelist(self):
         )
         default_for_country = pls and pls[0]
         actual = self.env['ir.property'].get('property_product_pricelist', 'res.partner', 'res.partner,%s' % self.id)
-
+        # We want to keep the current pricelist if it's still valid for the new country
+        # or if the country would be the same, like in the case of an state change that
+        # triggers this method and could lead to loose the original pricelist
+        update_country = False
+        if default_for_country and actual and default_for_country.id != actual.id:
+            if not actual.country_group_ids:
+                update_country = default_for_country.currency_id == actual.currency_id
+            else:
+                update_country = self.country_id not in actual.country_group_ids.country_ids
```

Comment:
If I'm not wrong, this could give singleton errors easily:

```suggestion
                update_country = self.country_id not in actual.mapped("country_group_ids.country_ids")
```

Reply:
True. Thanks ;)